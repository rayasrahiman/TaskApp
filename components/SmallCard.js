import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import AntIcon from 'react-native-vector-icons/AntDesign';
import EntypoIcon from 'react-native-vector-icons/Entypo';

import {Colors} from '../constants/colors';
import I18n from '../languages/i18n';

export default function SmallCard() {
  return (
    <View style={styles.container}>
      <View style={styles.cardWrapper}>
        <AntIcon style={styles.iconStyle} name="medicinebox" />
        <Text style={styles.cardTitle}>{I18n.t("IntelliConnect")}</Text>
        <Text style={styles.cardSubTitle}>Auto</Text>
      </View>
      <View style={styles.cardWrapper}>
        <EntypoIcon style={styles.iconStyle} name="cycle" />
        <Text style={styles.cardTitle}>{I18n.t("IntelliFlo")}</Text>
        <Text style={styles.cardSubTitle}>{I18n.t("ONuntil")}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginHorizontal: wp(5),
  },
  cardWrapper: {
    marginRight: wp(2),
    borderWidth: hp(0.3),
    borderColor: Colors.gray,
    borderRadius: hp(2),
    width: wp(44),
    height: hp(20),
  },
  iconStyle: {
    fontSize: hp(4),
    color: Colors.lightBlue,
    paddingHorizontal: wp(3),
    paddingTop: hp(2),
  },
  cardTitle: {
    color: Colors.lightBlue,
    fontWeight: 'bold',
    fontSize: hp(3.5),
    paddingHorizontal: wp(3),
    paddingTop: hp(2),
  },
  cardSubTitle: {
    color: Colors.lightBlue,
    fontSize: hp(3),
    paddingHorizontal: wp(3),
  },
});
