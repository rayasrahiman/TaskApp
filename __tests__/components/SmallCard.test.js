import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import SmallCard from '../../components/SmallCard';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('SmallCard', () => {
  test('should render SmallCard correctly', () => {
    const wrapper = render(<SmallCard />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
