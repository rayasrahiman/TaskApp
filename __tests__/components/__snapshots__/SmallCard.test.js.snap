// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallCard should render SmallCard correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginHorizontal": 37.5,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#CCCCCC",
        "borderRadius": 26.5,
        "borderWidth": 4,
        "height": 267,
        "marginRight": 15,
        "width": 330,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#0E6E96",
            "fontSize": 53.5,
            "paddingHorizontal": 22.5,
            "paddingTop": 26.5,
          },
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#0E6E96",
          "fontSize": 46.5,
          "fontWeight": "bold",
          "paddingHorizontal": 22.5,
          "paddingTop": 26.5,
        }
      }
    >
      IntelliConnect
    </Text>
    <Text
      style={
        Object {
          "color": "#0E6E96",
          "fontSize": 40,
          "paddingHorizontal": 22.5,
        }
      }
    >
      Auto
    </Text>
  </View>
  <View
    style={
      Object {
        "borderColor": "#CCCCCC",
        "borderRadius": 26.5,
        "borderWidth": 4,
        "height": 267,
        "marginRight": 15,
        "width": 330,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#0E6E96",
            "fontSize": 53.5,
            "paddingHorizontal": 22.5,
            "paddingTop": 26.5,
          },
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#0E6E96",
          "fontSize": 46.5,
          "fontWeight": "bold",
          "paddingHorizontal": 22.5,
          "paddingTop": 26.5,
        }
      }
    >
      IntelliFlo
    </Text>
    <Text
      style={
        Object {
          "color": "#0E6E96",
          "fontSize": 40,
          "paddingHorizontal": 22.5,
        }
      }
    >
      ON until 10pm
    </Text>
  </View>
</View>
`;
